### DISCUSSION TITLE
Can't Deploy my Smart Contract to Testnet

### DISCUSSION THREAD

**Discussion by saj404** [TENK-DAO/tenk#73] [Q&A] [answered]
I am using the yarn command for deploying the Smart Contract but for some reason, it says that the `accountId is undefined` even though I have logged in using the command `near login`. Is there a step on how to login my account through yarn to do this? Or is there other reason why it states that my accountId is undefined when using cli to deploy a Smart Contract?

**Comment by willemneal** [comment, marked as answer]
You need to provide the account id you want to deploy to.  e.g. `yarn deploy:testnet --acountId bob.testnet`

**Reply by saj404** [reply]
Thank you. I did this and it was able to deploy the Smart Contract. I encountered a problem though when I used the Tenk-Frontend-Starter using the smart contract I just deployed it gives me an error that "The contract is not initialized". I don't see other options in the package.json on initialization. Do I need to change other values aside from the value in `tenk.InitialMetadata` in deploy.ts?

**Reply by willemneal** [reply]
Hmm do you have a link to the transaction of when you deployed?

You need to provide all the arguments in `new_default_metadata`, which is at the bottom of that script.  Besides the initial metadata you mention this includes the owner and the size of the collection.

**Reply by saj404** [reply]
I solved it, I forgot to use the `new_default_metadata` you mentioned. Thank you!
